File: src/main/java_ex/vladsaif/syncedit/plugin/lang/transcript/TranscriptFileType.java
package vladsaif.syncedit.plugin.lang.transcript;

import com.intellij.icons.AllIcons;
import com.intellij.openapi.fileTypes.LanguageFileType;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

import javax.swing.*;

public class TranscriptFileType extends LanguageFileType {
    private static final TranscriptFileType INSTANCE = new TranscriptFileType();

    public static TranscriptFileType getInstance() {
        return INSTANCE;
    }

    private TranscriptFileType() {
        super(TranscriptLanguage.getInstance());
    }

    @NotNull
    @Override
    public String getName() {
        return "Audio transcript";
    }

    @NotNull
    @Override
    public String getDescription() {
        return "Audio transcript myPsiFile";
    }

    @NotNull
    @Override
    public String getDefaultExtension() {
        return "transcript";
    }

    @Nullable
    @Override
    public Icon getIcon() {
        return AllIcons.FileTypes.Text;
    }
}

File: src/main/kotlin/vladsaif/syncedit/plugin/lang/transcript/psi/TranscriptFileType.kt
package vladsaif.syncedit.plugin.lang.transcript.psi

import com.intellij.icons.AllIcons
import com.intellij.openapi.fileTypes.LanguageFileType
import java.util.*
import javax.swing.Icon

// Do not create file of this file type manually
object TranscriptFileType : LanguageFileType(TranscriptLanguage) {
  private val extension = UUID.randomUUID().toString()

  override fun getName() = "Transcript file"

  override fun getDescription() = "Transcript PSI Skeleton"

  override fun getDefaultExtension() = extension

  override fun getIcon(): Icon? = AllIcons.FileTypes.Text
}